src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -26,12 +35,39 @@ void handle_task_round_robin(global_scheduler_state *state,
                              global_scheduler_policy_state *policy_state,
                              task *task) {
   CHECKM(utarray_len(state->local_schedulers) > 0,
-         "No local schedulers. We currently don't handle this case.")
-  local_scheduler *scheduler = (local_scheduler *) utarray_eltptr(
-      state->local_schedulers, policy_state->round_robin_index);
-  policy_state->round_robin_index += 1;
-  policy_state->round_robin_index %= utarray_len(state->local_schedulers);
-  assign_task_to_local_scheduler(state, task, scheduler->id);
+         "No local schedulers. We currently don't handle this case.");
+  local_scheduler *ls = NULL;
+  task_spec *task_spec = task_task_spec(task);
+  int i;
+  int num_retries = NUM_RETRIES;
+  bool task_satisfied = false;
+
+  for (i = policy_state->round_robin_index;
```

Comment:
`ls->info.static_resources` doesn't change between iterations, right? Can we limit the number of rounds to the minimum of the number of local schedulers and `NUM_RETRIES`?